dotCMS/src/main/java/com/dotmarketing/business/FutureContentletPublishStateException.java

```diff
@@ -0,0 +1,21 @@
+package com.dotmarketing.business;
+
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+
+/**
+ * Throw when try to publish a page with a content with a future publish date
+ */
+public class FutureContentletPublishStateException extends PublishStateException{
+
+    private Contentlet contentlet;
+
+    public FutureContentletPublishStateException(final Contentlet contentlet) {
+        super("The content cannot be published because it is scheduled to be published on future date.");
```

Comment:
Add a property that would allow users/devs to identify what specific Contentlet was not published.

Reply:
already exists

Reply:
done https://github.com/dotCMS/core/pull/19988/commits/30658f7295ec9788fbc12fcd61c2f3b3bf07e272